Wikipedia/Code/WMFAppViewController.m

```diff
@@ -855,23 +855,30 @@ - (void)resumeApp:(dispatch_block_t)completion {
         };
 
         if (self.notificationUserInfoToShow) {
+            [self hideSplashViewAnimated:!didShowOnboarding];
             [self showInTheNewsForNotificationInfo:self.notificationUserInfoToShow];
             self.notificationUserInfoToShow = nil;
             done();
         } else if (self.unprocessedUserActivity) {
+            [self hideSplashViewAnimated:!didShowOnboarding];
             [self processUserActivity:self.unprocessedUserActivity animated:NO completion:done];
         } else if (self.unprocessedShortcutItem) {
+            [self hideSplashViewAnimated:!didShowOnboarding];
             [self processShortcutItem:self.unprocessedShortcutItem
                            completion:^(BOOL didProcess) {
                                done();
                            }];
-        } else if ([self shouldShowLastReadArticleOnLaunch]) {
-            [self showLastReadArticleAnimated:NO];
+        } else if (NSUserDefaults.wmf.shouldRestoreNavigationStackOnResume) {
+            [self.navigationStateController restoreNavigationStateFor:self.navigationController in:self.dataStore.viewContext with:self.theme completion:^{
+                [self hideSplashViewAnimated:!didShowOnboarding];
+            }];
             done();
```

Comment:
`done();` should probably move into the completion as well - would ensure two login requests aren't kicked off at the same time